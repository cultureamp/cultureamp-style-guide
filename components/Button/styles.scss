@import '~cultureamp-style-guide/styles/border';
@import '~cultureamp-style-guide/styles/color';
@import '~cultureamp-style-guide/styles/type';

%ca-button {
  display: inline-block;
  background: add-tint($ca-palette-ink, 80%);
  color: rgba($ca-palette-ink, 0.3);
  padding: 14px 21px;
  height: 47px;
  font-weight: $ca-weight-semibold;
  font-size: 14px;
  font-family: $ca-default-font-family;
  border-radius: 3px;
  transition: none;
  border: 0;
  outline: none;

  svg {
    position: absolute;
    top: 7px;
    width: 34px;
    height: 34px;
    color: inherit;
    line-height: 23px;
  }

  // We can't just used :enabled here, because we want to be able to apply these
  // styles to non-form elements, such as hyperlinks. :not(:disabled) is
  // equivalent on form elements, but also allows these styles to be applied to
  // elements that have no enabled/disabled states.
  &:not(:disabled) {
    background: $ca-palette-coral;
    color: #fff;
    cursor: pointer;

    &:hover,
    &:focus {
      background: add-shade($ca-palette-coral, 5%);
      text-decoration: none;
      color: #fff;
    }

    &:active {
      position: relative;
      top: 1px;
    }
  }
}

// Variations

%ca-button--with-icon {
  @extend %ca-button;
  padding-left: 50px;
  position: relative;

  svg {
    left: 14px;
  }
}

%ca-button--dropdown {
  // Put an =icon('icon-chevron-down') in the button for this one
  @extend %ca-button;
  padding-right: 63px;
  position: relative;
  width: 100%;
  text-align: left;

  svg {
    right: 10px;
  }
}

// Modifiers

%ca-button---positive {
  // We can't just used :enabled here, because we want to be able to apply these
  // styles to non-form elements, such as hyperlinks. :not(:disabled) is
  // equivalent on form elements, but also allows these styles to be applied to
  // elements that have no enabled/disabled states.
  &:not(:disabled) {
    background: add-tint($ca-palette-seedling, 20%);

    &:hover,
    &:focus {
      background: $ca-palette-seedling;
    }

    &:active {
      background: add-shade($ca-palette-seedling, 5%);
    }
  }
}
