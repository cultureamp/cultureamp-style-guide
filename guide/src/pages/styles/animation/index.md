---
imports:
  AnimationDroplet: ./_AnimationDroplet.js
  AnimationDropletGrid: ./_AnimationDropletGrid.js
  IntroParagraph: components/IntroParagraph.js
  "{TipContainer,TipCard}": components/tip-card
---

# Animation

<IntroParagraph>

Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec iaculis lacus ac odio pretium, sed pellentesque quam commodo. In hac habitasse platea dictumst. Curabitur tempor, nisi et elementum facilisis, nibh erat laoreet neque, quis bibendum nunc leo eu dui.

</IntroParagraph>

## What is animation for?

* Create a consistent experiential 'feel' across the product.
* Provide the user with visual cues for direction and focus.
* ...

## Motion Design Principles

* Easing / timing functions:
* Duration & delays
* Stagger /& synchronise
* ...

## Glossary of Animation Presets

<AnimationDropletGrid>
  <AnimationDroplet type="pulse" color="#45AD8F" />

  <AnimationDroplet type="rubberband" color="#1B7688" />

  <AnimationDroplet type="jello" color="#253C64" />

  <AnimationDroplet type="fadeInUp" color="#727193" />

  <AnimationDroplet type="bounceInUp" color="#F3786D" />
</AnimationDropletGrid>

## Technical Considerations

* Achieving 60 fps
* Collapsing height
* ...

### Do’s and Dont’s

<TipContainer>
<TipCard title="Icons should…" type="tip">

* use CSS transformations.

</TipCard>
<TipCard title="Icons should not…" type="warning">

* avoid animating width, height... properties where possible

</TipCard>
</TipContainer>
