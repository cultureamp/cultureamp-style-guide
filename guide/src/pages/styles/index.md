---
imports:
  Link: components/Link.js
---

# Visuals

> Elements of visual design that we seek to use consistently.

Menu:

* <Link to="/styles/colors">Colors</Link>
* <Link to="/styles/icons">Icons</Link>
* <Link to="/styles/typography">Typography</Link>
