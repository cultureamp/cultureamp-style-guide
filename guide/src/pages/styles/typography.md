---
imports:
  Link: components/Link.js
  IntroParagraph: components/IntroParagraph.js
  "{TipContainer,TipCard}": components/tip-card
---

# Typography

<IntroParagraph>

Typography is important when establishing a strong design system. When established, it can help to create balance and hierarchy of textual information. By creating structure within a page, text becomes enjoyable and appealing to read. We aim to ensure type within our product complies with WCAG 2.0 AA standards.

</IntroParagraph>

## What is its purpose?

* Create visual hierarchy of textual information

## Why is it useful?

* Makes reading text enjoyable / appealing / legible
* Creates structure within a page

## How should we use it?

* Comply with WCAG 2.0 AA standards (12px minimum, 16px / 1em standard)
* Utilise brand typefaces (Open Sans → Ideal Sans (Sans Serif) and Monaco → Operator (Monospace))
* Text colour / contrast is crucial. Ensure colour ratios comply with WCAG 2.0 AA
* Adhere to our baseline grid (line-height) of 24px / Vertical rhythm
* Ensure paragraph widths are not too long (650px wide max or 90 characters)
* White space is equally important. Let your text breathe.
* Use type styles to define content structure (Page titles, Intro Lede, Body, Small etc)
* Render correctly (i.e. not janky) across all devices and operating systems.

## How shouldn’t we use it?

* You can also view the <Link to='/language/checklist'>Product Writing Checklist</Link> to quickly check any text you are adding to the Culture Amp product or related communications.

## General guidelines

* Basic structure of hierarchy (examples)
* Paragraph spacing / Vertical rhythm
* Paragraph width
* Links (when to underline, when not to)
* How to style Abbreviations (E.g. HRIS) / Underline / Tooltips

## Best practices

<TipContainer>
<TipCard title="Typography should…" type="tip">

* A pleasure to read

* Should make sense, visually and informationally

</TipCard>
<TipCard title="Typography should not…" type="warning">

* Be overwhelming

* Difficult to read

</TipCard>
</TipContainer>
