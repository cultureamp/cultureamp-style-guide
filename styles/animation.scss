// Timing variables
// These timing values are designed to snap together. 100+300=400. 300+400=700 etc.

$ca-duration-instant: 0ms;
$ca-duration-immediate: 100ms;
$ca-duration-fast: 300ms;
$ca-duration-slow: 400ms;
$ca-duration-deliberate: 700ms;
