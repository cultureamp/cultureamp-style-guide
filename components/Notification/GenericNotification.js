// @flow
import * as React from 'react';
import Icon from '../Icon/Icon';
import successIcon from '../../icons/success.svg';
import exclamationIcon from '../../icons/exclamation.svg';
import infoIcon from '../../icons/information.svg';
import closeIcon from '../../icons/close.svg';
import styles from './Notification.module.scss';
import classnames from 'classnames';

type Props = {|
  type: 'positive' | 'warning' | 'negative' | 'info',
  style: 'global' | 'normal' | 'toast',
  persistent?: boolean,
  title?: string,
  children?: ?React.Node,
  autohide?: ?boolean,
  onHide?: ?() => void | any,
|};

type State = {
  isFading: boolean,
};

class GenericNotification extends React.Component<Props, State> {
  state = {
    isFading: true,
  };

  componentDidMount() {
    requestAnimationFrame(() => this.setState({ isFading: false }));

    if (this.props.style === 'toast' && this.props.autohide) {
      setTimeout(() => this.triggerHide(), 5000);
    }
  }

  render() {
    return (
      <div className={this.determineClassName()}>
        <div className={styles.icon}>
          <Icon icon={this.getIconType()} role="presentation" inheritSize />
        </div>
        <div className={styles.textContainer}>
          {this.props.title && (
            <span className={styles.title}>{this.props.title}</span>
          )}
          <span className={styles.textWrapper}>
            <span className={styles.text}>{this.props.children}</span>
          </span>
        </div>
        {!this.props.persistent && this.renderCancelButton()}
      </div>
    );
  }

  determineClassName(): string {
    return classnames(
      styles.notification,
      styles[this.props.type],
      styles[this.props.style],
      {
        [styles.fading]: this.state.isFading,
      }
    );
  }

  getIconType() {
    switch (this.props.type) {
      case 'positive':
        return successIcon;
      case 'negative':
        return exclamationIcon;
      case 'warning':
        return exclamationIcon;
      case 'info':
        return infoIcon;
      default:
        return infoIcon;
    }
  }

  renderCancelButton() {
    return (
      <span className={styles.cancel} onClick={() => this.triggerHide()}>
        <span className={styles.cancelInner}>
          <Icon
            icon={closeIcon}
            role="img"
            title="close notification"
            inheritSize
          />
        </span>
      </span>
    );
  }

  triggerHide() {
    requestAnimationFrame(() => this.setState({ isFading: true }));

    setTimeout(() => {
      this.props.onHide && this.props.onHide();
    }, 600);
  }
}

export default GenericNotification;
