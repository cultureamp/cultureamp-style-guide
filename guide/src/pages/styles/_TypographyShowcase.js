import React from 'react';
import styles from './_TypographyShowcase.module.scss';

class TypographyItem extends React.Component {
  state = {};

  componentDidMount() {
    let style = window.getComputedStyle(this.paragraph);
    this.setState({
      size: `${style.fontSize} / ${style.lineHeight}`,
    });
  }

  render() {
    let { name, sampleText, className, sassCode } = this.props;
    let title = this.state.size ? `${name} – ${this.state.size}` : name;
    return (
      <div className={styles.typographyShowcaseItem}>
        <h3>{title}</h3>
        <p ref={p => (this.paragraph = p)} className={styles[className]}>
          {sampleText}
        </p>
        <pre>{sassCode}</pre>
      </div>
    );
  }
}

const TypographyShowcase = () => (
  <div>
    <h2 className={styles.display}>Type hierarchy</h2>
    <TypographyItem
      name="Page Title (H1)"
      sampleText="Have the courage to be vulnerable"
      className="pageTitle"
      sassCode="@include ca-type-ideal-page-title;"
    />
    <TypographyItem
      name="Title (H2)"
      sampleText="Have the courage to be vulnerable"
      className="title"
      sassCode="@include ca-type-ideal-title;"
    />
    <TypographyItem
      name="Display (H3)"
      sampleText="Learn faster through feedback"
      className="display"
      sassCode="@include ca-type-ideal-display;"
    />
    <TypographyItem
      name="Heading (H4)"
      sampleText="Trust people to make decisions"
      className="heading"
      sassCode="@include ca-type-ideal-heading;"
    />
    <TypographyItem
      name="Lede (Intro)"
      sampleText="Values are not what you want to be on a good day. Values are what you are willing to hurt for every day."
      className="lede"
      sassCode="@include ca-type-ideal-lede;"
    />
    <TypographyItem
      name="Body"
      sampleText="Dr. Brené Brown, author of Daring Greatly, is a research professor from the University of Houston who studies human emotions, including shame and vulnerability. In a March 2012 TED talk, she said, “Vulnerability is not weakness, and that myth is profoundly dangerous.” She went on to say that after 12 years of research, she has actually determined that vulnerability is “our most accurate measurement of courage.”"
      className="body"
      sassCode="@include ca-type-ideal-body;"
    />
    <TypographyItem
      name="Body Bold"
      sampleText="Our mission is to make the power of people analytics accessible to everyone."
      className="bodyBold"
      sassCode="@include ca-type-ideal-body-bold;"
    />
    <TypographyItem
      name="Small"
      sampleText="Had I the heaven's embroidered cloths. Enwrought with golden and silver light. The blue and the dim and the dark cloths of night and light and the half-light. I would spread the cloths under your feet. But I, being poor, have only my dreams. I have spread my dreams under your feet. Tread softly because you tread on my dreams."
      className="small"
      sassCode="@include ca-type-ideal-small;"
    />
    <TypographyItem
      name="Small Bold"
      sampleText="Had I the heaven's embroidered cloths. Enwrought with golden and silver light. The blue and the dim and the dark cloths of night and light and the half-light. I would spread the cloths under your feet. But I, being poor, have only my dreams. I have spread my dreams under your feet. Tread softly because you tread on my dreams."
      className="smallBold"
      sassCode="@include ca-type-ideal-small-bold;"
    />
    <TypographyItem
      name="Pre"
      sampleText="@import 'cultureamp-style-guide/styles/type';"
      className="pre"
      sassCode="@include ca-type-pre;"
    />
    <TypographyItem
      name="Pre Bold"
      sampleText="@import 'cultureamp-style-guide/styles/color';"
      className="preBold"
      sassCode="@include ca-type-pre-bold;"
    />
    <TypographyItem
      name="Notifications"
      sampleText={
        <span>
          "Had I the heaven's embroidered cloths.
          <br />Enwrought with golden and silver light.
          <br />The blue and the dim and the dark cloths of night and light and
          the half-light.
          <br />I would spread the cloths under your feet.
          <br />But I, being poor, have only my dreams.
          <br />I have spread my dreams under your feet.
          <br />Tread softly because you tread on my dreams."
          <br />&mdash; William Butler Yeats
        </span>
      }
      className="notification"
      sassCode="@include ca-type-ideal-notification;"
    />
    <TypographyItem
      name="Labels"
      sampleText="Fact &amp; Truth"
      className="label"
      sassCode="@include ca-type-ideal-label;"
    />
    <TypographyItem
      name="Control Action"
      sampleText="Fact &amp; Truth"
      className="controlAction"
      sassCode="@include ca-type-ideal-control-action;"
    />
    <TypographyItem
      name="Buttons"
      sampleText="Amplify Others"
      className="button"
      sassCode="@include ca-type-ideal-button;"
    />
  </div>
);

export default TypographyShowcase;
