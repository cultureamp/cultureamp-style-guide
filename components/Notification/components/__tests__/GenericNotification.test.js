import React from 'react';
import { mount } from 'enzyme';
import sinon from 'sinon';
import GenericNotification from '../GenericNotification';
import styles from '../GenericNotification.module.scss';

beforeEach(() => {
  jest.useFakeTimers();
});

test('Begins "hidden" but transitions out of it immediately', () => {
  const notification = mount(
    <GenericNotification type="positive" style="inline" title="Success!">
      This is my positive notification
    </GenericNotification>
  );

  expect(notification.childAt(0).hasClass(styles.hidden)).toBe(true);
  jest.advanceTimersByTime(1);
  notification.update();
  expect(notification.childAt(0).hasClass(styles.hidden)).toBe(false);
});

test('The cancel button hides the notification and triggers the onHide callback', () => {
  const onHide = sinon.spy();
  const notification = mount(
    <GenericNotification
      type="positive"
      style="inline"
      title="Success!"
      onHide={onHide}
    >
      This is my positive notification
    </GenericNotification>
  );

  // The element should start in a "hidden" state
  expect(notification.state('hidden')).toBe(true);
  expect(notification.state('removed')).toBe(false);
  expect(onHide.callCount).toBe(0);

  // It should immediately start transitioning into a visible state
  jest.advanceTimersByTime(1);
  notification.update();
  expect(notification.state('hidden')).toBe(false);
  expect(notification.state('removed')).toBe(false);
  expect(onHide.callCount).toBe(0);
  expect(notification).toMatchSnapshot();

  // After clicking, the element should fade out, but the onHide not trigger yet.
  notification.find('.' + styles.cancel).simulate('click');
  jest.advanceTimersByTime(1);
  notification.update();
  expect(notification.state('hidden')).toBe(true);
  expect(notification.state('removed')).toBe(false);
  expect(onHide.callCount).toBe(0);
  expect(notification).toMatchSnapshot();

  // After the fade out animation has finished, the onHide handler should trigger.
  notification.children().simulate('transitionend', {
    propertyName: 'margin-top',
  });
  notification.update();
  expect(notification.state('hidden')).toBe(true);
  expect(notification.state('removed')).toBe(true);
  expect(onHide.callCount).toBe(1);
  expect(notification).toMatchSnapshot();
});

test('If autohide is specified, we should start hiding after 5s', () => {
  const notification = mount(
    <GenericNotification
      type="positive"
      style="toast"
      title="Success!"
      autohide
    >
      This is my positive notification
    </GenericNotification>
  );

  // After 4s, it should still be visible
  jest.advanceTimersByTime(4999);
  notification.update();
  expect(notification.state('hidden')).toBe(false);

  // After 5s, it should be fading out to "hidden"
  jest.advanceTimersByTime(1);
  notification.update();
  expect(notification.state('hidden')).toBe(true);
});
