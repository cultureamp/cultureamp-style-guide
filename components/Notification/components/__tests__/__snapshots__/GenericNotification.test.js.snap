// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The cancel button hides the notification and triggers the onHide callback 1`] = `
<GenericNotification
  autohide={false}
  onHide={[Function]}
  persistent={false}
  style="inline"
  title="Success!"
  type="positive"
>
  <div
    className="notification positive inline"
    onTransitionEnd={[Function]}
    style={
      Object {
        "marginTop": undefined,
      }
    }
  >
    <div
      className="icon"
    >
      <Icon
        icon={null}
        inheritSize={true}
        role="presentation"
      >
        <svg
          aria-hidden={true}
          className="icon inheritSize"
          focusable="false"
          role="presentation"
          viewBox="viewBox"
        >
          <use
            xlinkHref="#id"
          />
        </svg>
      </Icon>
    </div>
    <div
      className="textContainer"
    >
      <h6
        className="title"
      >
        Success!
      </h6>
      <p
        className="text"
      >
        This is my positive notification
      </p>
    </div>
    <CancelButton
      onClick={[Function]}
    >
      <button
        className="cancel"
        onClick={[Function]}
      >
        <span
          className="cancelInner"
        >
          <Icon
            icon={null}
            inheritSize={false}
            role="img"
            title="close notification"
          >
            <svg
              aria-labelledby="id-4-title"
              className="icon"
              focusable="false"
              role="img"
              viewBox="viewBox"
            >
              <title
                id="id-4-title"
              >
                close notification
              </title>
              <use
                xlinkHref="#id"
              />
            </svg>
          </Icon>
        </span>
      </button>
    </CancelButton>
  </div>
</GenericNotification>
`;

exports[`The cancel button hides the notification and triggers the onHide callback 2`] = `
<GenericNotification
  autohide={false}
  onHide={[Function]}
  persistent={false}
  style="inline"
  title="Success!"
  type="positive"
>
  <div
    className="notification positive inline hidden"
    onTransitionEnd={[Function]}
    style={
      Object {
        "marginTop": "NaNpx",
      }
    }
  >
    <div
      className="icon"
    >
      <Icon
        icon={null}
        inheritSize={true}
        role="presentation"
      >
        <svg
          aria-hidden={true}
          className="icon inheritSize"
          focusable="false"
          role="presentation"
          viewBox="viewBox"
        >
          <use
            xlinkHref="#id"
          />
        </svg>
      </Icon>
    </div>
    <div
      className="textContainer"
    >
      <h6
        className="title"
      >
        Success!
      </h6>
      <p
        className="text"
      >
        This is my positive notification
      </p>
    </div>
    <CancelButton
      onClick={[Function]}
    >
      <button
        className="cancel"
        onClick={[Function]}
      >
        <span
          className="cancelInner"
        >
          <Icon
            icon={null}
            inheritSize={false}
            role="img"
            title="close notification"
          >
            <svg
              aria-labelledby="id-4-title"
              className="icon"
              focusable="false"
              role="img"
              viewBox="viewBox"
            >
              <title
                id="id-4-title"
              >
                close notification
              </title>
              <use
                xlinkHref="#id"
              />
            </svg>
          </Icon>
        </span>
      </button>
    </CancelButton>
  </div>
</GenericNotification>
`;

exports[`The cancel button hides the notification and triggers the onHide callback 3`] = `
<GenericNotification
  autohide={false}
  onHide={[Function]}
  persistent={false}
  style="inline"
  title="Success!"
  type="positive"
/>
`;
